xbmc/music/MusicLibraryQueue.cpp

```diff
@@ -73,12 +73,12 @@ void CMusicLibraryQueue::ExportLibrary(const CLibExportSettings& settings, bool
 
     if (progress)
     {
-      // Render and wait for export to complete or be cancelled
+      // Render (so pointer movements work on dialog) and wait for export to complete or be cancelled
       while (progress->IsActive() && !progress->IsCanceled())
+      {
         progress->Progress();
-      // Finally close progress dialog
-      if (progress->IsActive())
-        progress->Close();
+        Sleep(10);
```

Comment:
Is this a microsecond, millisecond or second sleep?

Reply:
A somewhat arbitrary 10 millisecond, but fails to compile on anything but the Windows platform, so there is something about sleep I do not know.

The intent here is to have the progress dialog rendering on a separate thread from the export process, otherwise the pointer movement is stilted and you can't get to the cancel button. Hence polling progress() calls which gives the dialog chance to update and render mouse pointer.

Reply:
#ifdef TARGET_POSIX
#include "linux/XTimeUtils.h"
#endif